src/main/kotlin/org/wycliffeassociates/otter/jvm/recorder/app/viewmodel/RecorderViewModel.kt

```diff
@@ -0,0 +1,107 @@
+package org.wycliffeassociates.otter.jvm.recorder.app.viewmodel
+
+import javafx.animation.AnimationTimer
+import javafx.application.Platform
+import javafx.beans.binding.BooleanBinding
+import javafx.beans.property.SimpleBooleanProperty
+import javafx.beans.property.SimpleStringProperty
+import org.wycliffeassociates.otter.common.app.ActiveRecordingRenderer
+import org.wycliffeassociates.otter.common.app.RecordingTimer
+import org.wycliffeassociates.otter.common.app.WavFileWriter
+import org.wycliffeassociates.otter.common.io.wav.WavFile
+import org.wycliffeassociates.otter.jvm.recorder.app.view.drawables.BaseWaveLine
+import org.wycliffeassociates.otter.jvm.recorder.app.view.CanvasFragment
+import org.wycliffeassociates.otter.jvm.recorder.app.view.FramerateView
+import org.wycliffeassociates.otter.jvm.recorder.app.view.drawables.WaveformLayer
+import org.wycliffeassociates.otter.jvm.recorder.device.AudioRecorder
+import org.wycliffeassociates.otter.jvm.recorder.app.view.drawables.VolumeBar
+import tornadofx.ViewModel
+import tornadofx.add
+import tornadofx.getValue
+import tornadofx.setValue
+import java.io.File
+
+
+class RecorderViewModel : ViewModel() {
+
+    val wav = WavFile(File(app.parameters.named["wav"]))
+    val recorder = AudioRecorder()
+    var volumeTest: AudioRecorder? = AudioRecorder()
```

Comment:
What is the purpose of volumeTest? It doesn't seem like the user interacts with it?

Reply:
when this app is launched, the microphone is immediately live so as to show a volume meter to give a rough idea of how loud or quiet they will be, but also helps in hinting at issues such as the wrong microphone being active, or none at all.

Once they start recording it goes away, and switches to a state of writing the audio to a file.